external-crates/move/crates/move-compiler/src/typing/translate.rs

```diff
@@ -2937,12 +2940,82 @@ fn module_call_impl(
     (call, return_)
 }
 
+/// If the constant that we are referencing has an `error` attribute, we need to change the type of
+/// the constant to a u64 since this will be compiled into a u64 error code.
+fn annotated_error_const(context: &mut Context, e: &mut T::Exp) {
+    if let sp!(
+        const_loc,
+        T::UnannotatedExp_::Constant(module_ident, constant_name)
+    ) = &mut e.exp
+    {
```

Comment:
`let sp!(_, Constant(...)) = &mut e.exp else return` might be better here

Reply:
Was on the fence between that and this -- will change :)